### PR TITLE
context : remove logits_all flag

### PR DESCRIPTION
**Pull request by ggerganov**
This flag has been deprecated for quite some time and in order to simplify the logic in `llama_context` it's better to remove it. To achieve the same functionality, set `llama_batch::logits` accordingly.

### `libllama` API Changes

- Remove `logits_all` flag
- Move the `bool` flags at the end of the struct

### CODE REVIEW

**Review by slaren — COMMENTED** [review]
I think it still needs to be removed from `llama_context_params`.

**Comment by ggerganov** [comment]
Removed the flag from `llama_context_params` and also moved the `bools` to the end of the struct. Will update the API changelog with this breaking change.

**Review by slaren — APPROVED** [review]
